packages/desktop-client/src/components/budget/misc.js

```diff
@@ -943,15 +1039,28 @@ const BudgetCategories = memo(
       let items = Array.prototype.concat.apply(
         [],
         expenseGroups.map(group => {
-          let items = [{ type: 'expense-group', value: group }];
+          if (group.hidden && !showHiddenCategories) {
+            return [];
+          }
+
+          const groupCategories = group.categories.filter(
+            cat => showHiddenCategories || !cat.hidden,
+          );
+
+          // //checking group.categories.length to make sure groups with no categories still show
+          // if (group.categories.length > 0 && groupCategories.length === 0) {
+          //   return [];
+          // }
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 suggestion:** ‏can be removed